### ISSUE TITLE
X-Zone cursor alignment

### ISSUE THREAD

**Issue by Harrumph** [MiSTer-devel/SNES_MiSTer#301]
Hi, been using the lightgun capability with a Wiimote. In all games, crosshair is aligned with target (you can see bullets hit on the crosshair), and one can correctly target at both right and left edges of the screen. However, like mentioned in a recent MLiG video, in the game X-zone the crosshair is mis-aligned, always to the right of the in-game cursor/bullets by maybe 10-20 pixels or so. This seems to lead to the cursor cannot be brought fully to the right edge of the screen, while it can on the left side. 
Unfortunately, I can’t test if this is the same behavior as on original hardware.

**Comment by paulb-nl** [comment]
I have tested the Scope on original hardware with X-Zone and T2 - The Arcade Game and there the cursor seems to be properly centered to the barrel. 

The simulated lightgun does not take into account the signal delay of the Scope which is about 18 pixels (40-22) according to https://problemkaputt.de/fullsnes.htm#snespputimersandstatus

So here is a build that delays the signal of the lightgun. The game cursor is now much closer to the core cursor.
[SNES_scope.zip](https://github.com/MiSTer-devel/SNES_MiSTer/files/7576177/SNES_scope.zip)

Another difference that I found is that on T2 the game cursor is only updated when pressing a button on the Scope. This is probably a battery saving measure. On the simulated lightgun the cursor is always updated.

**Comment by Harrumph** [comment]
Thanks a lot Paul, I’ll try to test your build in the coming days! Just to clarify, it was possible on original hardware to move the cursor all the way to the right edge of the screen in X-zone?

**Comment by paulb-nl** [comment]
Yes in X-Zone on an original SNES the cursor is able to move half off screen in both directions. So this means the middle of the cursor is at the edge.

**Comment by Harrumph** [comment]
I have tested your build and it does indeed align the core cursor with the game cursor much, much better!
The timing is not quite perfect as the cursor cannot be moved half off screen on right side, only to the edge of the cursor (on the left it is moved half off though). But the game play experience is greatly enhanced already.
Let me know if more testing is needed before commit. Thanks a lot!

**Comment by paulb-nl** [comment]
I checked the Scope again. Here is a screenshot with the cursor at both outer edges with a real Scope through SNAC.
![20211124_164953-X Zone (Japan, USA)_LR](https://user-images.githubusercontent.com/2227984/143300463-1d58e1a9-4db2-4826-bf91-1ca0a321af95.png)

I added some more delay to the simulated scope to make it as close as possible for the left and right edges. There is only 1 or 2 pixels difference now and the core cursor is exactly in the middle of the  game cursor horizontally.
[SNES_scope2.zip](https://github.com/MiSTer-devel/SNES_MiSTer/files/7598248/SNES_scope2.zip)
